andela/ah-magnificent6
# #159965312 Favourite and unfavourite articles

**What does this PR do?**
Enable user to favourite and unfavourite an article

**Description of Task to be completed?**
Articles that tickle our fancy are worth favouriting and those deemed distasteful are worth unfavoriting.

**How should this be manually tested?**
$ git clone https://github.com/andela/ah-magnificent6.git
$ git checkout ft-favourite-article-159965312
$ cd ah-magnificent6
$ cp .env-sample .env [and set the environment variables]
$ python3 -m venv venv
$ source venv/bin/activate
(venv) $ pip install -r requirements.txt
(venv) $ python manage.py migrate
(venv) $ python manage.py runserver

Using [Postman](https://www.getpostman.com/) navigate to 

> {your-server-root}:8000

The expected payloads are in the format:
User registration: `POST /api/user/signup`
```
{
    "email": "author@user.user",
    "username": "author",
    "password": "goodpassword"
}
```

User Login: `POST /api/user/login`
```
{
    "email": "author@user.user",
    "password": "goodpassword"
}
```
Use the token from the response in the login as the authentication header while creating an article.

<img width="1388" alt="screen shot 2018-09-12 at 10 54 16" src="https://user-images.githubusercontent.com/39065274/45410426-500b8e00-b67a-11e8-91ed-49338b9f3ab3.png">

Create article : POST /api/articles/
```
{
"title": "My Journey to Andela",
 "description": "This article is about how I joined Andela",
"body": "This is my story to Andela"
}
```
From the response after creating an article, you will get a slug which you will use to favourite the article.

Favorite/unfavourite article: POST /api/articles/slug/favourite
<img width="1373" alt="screen shot 2018-09-12 at 10 27 14" src="https://user-images.githubusercontent.com/39065274/45410719-18e9ac80-b67b-11e8-8cbd-d7f0f366ae94.png">


<img width="1373" alt="screen shot 2018-09-12 at 10 26 17" src="https://user-images.githubusercontent.com/39065274/45410739-22731480-b67b-11e8-9db2-7114705ec485.png">


**What are the relevant pivotal tracker stories?**
[#159965312](https://www.pivotaltracker.com/story/show/159965312)
